demo_modules/glslsandbox/caramel.js

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2018 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+const register = require('register');
+const ModuleInterface = require('lib/module_interface');
+const wallGeometry = require('wallGeometry');
+
+class CaramelServer extends ModuleInterface.Server {}
+
+class CaramelClient extends ModuleInterface.Client {
+  constructor(config) {
+    super();
+  }
+
+  finishFadeOut() {
+    if (this.surface) {
+      this.surface.destroy();
+    }
+  }
+  
+  willBeShownSoon(container, deadline) {
+    const GlslSandboxSurface = require('client/surface/glslsandbox_surface');
+    
+    /*
+1. add this line
```

Comment:
what's up with this commented code?

Reply:
They're my instructions to myself for how to port from glslsandbox...  I keep copying a demo module and modifying it, and these are the steps necessary.

Reply:
perhaps that is better as part of the surface file, then?  We had a site for docs, but I forgot where it lives.  :)

Reply:
these are probably better off in the surface file, then.